core-bundle/src/Routing/Matcher/LegacyMatcher.php

```diff
@@ -192,19 +192,18 @@ private function parseSuffixAndLanguage(string $pathInfo, ?string &$locale): str
      */
     private function rebuildRequest(string $pathinfo, Request $request): Request
     {
-        $uri = $pathinfo;
-        $server = [];
+        $server = $request->server->all();
 
         if ($request->getBaseUrl()) {
-            $uri = $request->getBaseUrl().$pathinfo;
-            $server['SCRIPT_FILENAME'] = $request->getBaseUrl();
-            $server['PHP_SELF'] = $request->getBaseUrl();
+            $pathinfo = $request->getBaseUrl().$pathinfo;
         }
 
-        $host = $request->getHttpHost() ?: 'localhost';
-        $scheme = $request->getScheme() ?: 'http';
-        $uri = $scheme.'://'.$host.$uri.'?'.$request->getQueryString();
+        if ($request->getQueryString()) {
+            $pathinfo .= '?'.$request->getQueryString();
+        }
```

Comment:
```suggestion
        if ($qs = $request->getQueryString()) {
            $pathinfo .= '?'.$qs;
        }
```

Not sure if `$request->getQueryString()` is an expensive call, but this generally seems like a reasonable pattern to be.